pkg/detectors/typeform/v2/typeform.go

```diff
@@ -0,0 +1,96 @@
+package typeform
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct{}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	client = common.SaneHttpClient()
+
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat = regexp.MustCompile(`\btfp_[a-zA-Z0-9_\.=\-]{40,59}\b`)
+)
+
+func (s Scanner) Version() int { return 2 }
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"typeform", "tfp_"}
+}
+
+type TypeFormResponse struct {
+	UserID   string `json:"user_id,omitempty"`
+	Email    string `json:"email,omitempty"`
+	Alias    string `json:"alias,omitempty"`
+	Language string `json:"language,omitempty"`
+}
+
+// FromData will find and optionally verify Typeform secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	matches := keyPat.FindAllString(dataStr, -1)
+
+	if len(matches) == 0 {
+		return
+	}
+
+	for _, match := range matches {
+		s1 := detectors.Result{
+			DetectorType: detectorspb.DetectorType_Typeform,
+			Raw:          []byte(match),
+		}
+
+		if verify {
+			req, err := http.NewRequestWithContext(ctx, "GET", "https://api.typeform.com/me", nil)
```

Comment:
The older convention for detector logic has a bunch of issues. I'd recommend moving this into a separate function (for the sake of `defer`) and having explicit error handling + status code checks.

e.g., https://github.com/trufflesecurity/trufflehog/blob/abaacd7e8af3019d622848641a095e471b6daf74/pkg/detectors/alchemy/alchemy.go#L68